assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties

```diff
@@ -479,13 +479,31 @@ che.infra.kubernetes.installer_server_min_port=10000
 che.infra.kubernetes.installer_server_max_port=20000
 
 # Defines annotations for ingresses which are used for servers exposing. Value depends on ingress controller.
-# For example for nginx ingress controller 0.9.0-beta17 the following value is recommended:
-# {"ingress.kubernetes.io/rewrite-target": "/","ingress.kubernetes.io/ssl-redirect": "false",\
-#     "ingress.kubernetes.io/proxy-connect-timeout": "3600","ingress.kubernetes.io/proxy-read-timeout": "3600"}
 #
 # OpenShift infrastructure ignores this property because it uses Routes instead of ingresses.
+#
+# Note that for a single-host deployment strategy to work, a controller supporting URL rewriting has to be
+# used (so that URLs can point to different servers while the servers don't need to support changing the app root).
+# See the che.infra.kubernetes.ingress.path.rewrite_transform property for further details on how to achieve
+# this.
+#
+# For example for nginx ingress controller 0.22.0 and later the following value is recommended:
+# {"ingress.kubernetes.io/rewrite-target": "/$1","ingress.kubernetes.io/ssl-redirect": "false",\
+#     "ingress.kubernetes.io/proxy-connect-timeout": "3600","ingress.kubernetes.io/proxy-read-timeout": "3600"}
+# and the che.infra.kubernetes.ingress.path.rewrite_transform should be set to "%s(.*)"
+#
+# For nginx ingress controller older than 0.22.0, the rewrite-target should be set to merely "/" and the path transform
+# to "%s" (see the the che.infra.kubernetes.ingress.path.rewrite_transform property).
 che.infra.kubernetes.ingress.annotations_json=NULL
 
+# Defines a "recipe" on how to declare the path of the ingress that should expose a server.
+# The "%s" represents the base public URL of the server and is guaranteed to end with a forward slash. This property
+# must be a valid input to the String.format() method and contain exactly one reference to "%s".
+#
+# Please see the description of the che.infra.kubernetes.ingress.annotations_json property to see how these two
+# properties interplay when specifying the ingress annotations and path.
+che.infra.kubernetes.ingress.path_transform=NULL
```

Comment:
The `(.*)` part is specific to nginx ingress controller so I didn't want the generic description to contain a specific example. Also, this property only makes sense together with the setting of the ingress annotations. That's why there is the note to refer to the description of the `che.infra.kubernetes.ingress.annotations_json` property, which contains a complete example for nginx ingress controller. Although there still is no description what the `(.*)` means there, this becomes clear for users that set this up because they need to understand the configuration of the ingress controllers. The nginx ingress controller documentation explains how the ingress path and the rewrite_target cooperate.

Other ingress controller might not even have a concept of rewrite_target and thus would not be able to use these two properties.

I tried to remain brief in the property descriptions but maybe I will have to reformulate some things.

Do you have any suggestions?

Reply:
No that makes sense to me, I guessed it was probably me not understanding some details of deploying on Kubernetes anyways. Consider maybe adding a link to some k8s documentation, e.g. https://kubernetes.github.io/ingress-nginx/examples/rewrite/